### PR TITLE
Server connector error handling improvements

### PR DESCRIPTION
**Pull request by Kishanthan**
- This PR depends on https://github.com/wso2/carbon-transports/pull/220 and the release of carbon-transports 4.2.6

- This PR uses a similar approach proposed in https://github.com/ballerinalang/ballerina/pull/1865 to handle 404 status code, but since the error handler is now at bellerina level, creating http status code specific messages can be done within the error handler.

### CODE REVIEW

**Comment by Kishanthan** [comment]
@imesh , can you review this and see whether the fix you wanted to add will also be included?

File: modules/ballerina-core/src/main/java/org/ballerinalang/services/dispatchers/websocket/WebSocketResourceDispatcher.java [hunk 1]
```diff
@@ -69,7 +70,7 @@ public Resource findResource(Service service, CarbonMessage cMsg, CarbonCallback
             throw new BallerinaException("Error occurred in WebSocket resource dispatchers : " + e.getMessage(),
                                          balContext);
         }
-        throw new BallerinaException("No matching Resource found for dispatchers.");
+        throw new ResourceNotFoundException("No matching Resource found for dispatchers.");
```
Comment (imesh) [inline modules/ballerina-core/src/main/java/org/ballerinalang/services/dispatchers/websocket/WebSocketResourceDispatcher.java]: As I see in ServerConnectorMessageHandler.java:71 "service" has been written in lowercase. Shouldn't we follow the same here for "resource"?

File: modules/ballerina-core/src/main/java/org/ballerinalang/services/dispatchers/jms/JMSServiceDispatcher.java [hunk 2]
```diff
@@ -46,15 +47,16 @@
...
 
     @Override
-    public Service findService(CarbonMessage cMsg, CarbonCallback callback, Context balContext) {
+    public Service findService(CarbonMessage cMsg, CarbonCallback callback, Context balContext)
+            throws ServiceNotFoundException {
         Object serviceIdProperty = cMsg.getProperty(Constants.JMS_SERVICE_ID);
         String serviceId = (serviceIdProperty != null) ? serviceIdProperty.toString() : null;
         if (serviceId == null) {
-            throw new BallerinaException("Service Id is not found in JMS Message", balContext);
+            throw new ServiceNotFoundException("Service Id is not found in JMS Message", balContext);
```
Comment (imesh) [inline modules/ballerina-core/src/main/java/org/ballerinalang/services/dispatchers/jms/JMSServiceDispatcher.java]: As I see in ServerConnectorMessageHandler.java:71 "service" has been written in lowercase. Shouldn't we follow the same here for "Message"?

File: modules/ballerina-core/src/main/java/org/ballerinalang/services/dispatchers/http/HTTPErrorHandler.java [hunk 3]
```diff
@@ -38,11 +38,14 @@
         service = ServerConnectorErrorHandler.class)
 public class HTTPErrorHandler implements ServerConnectorErrorHandler {
 
-    private static final Logger log = LoggerFactory.getLogger(HTTPErrorHandler.class);
-
     @Override
-    public void handleError(Exception e, CarbonMessage carbonMessage, CarbonCallback callback) {
-        callback.done(createErrorMessage(e.getMessage(), 500));
+    public void handleError(Exception exception, CarbonMessage carbonMessage, CarbonCallback callback) {
+        if (exception.getCause() instanceof ServiceNotFoundException ||
```
Comment (imesh) [inline modules/ballerina-core/src/main/java/org/ballerinalang/services/dispatchers/http/HTTPErrorHandler.java]: This looks good! Wouldn't it also be better to check for the exception object similar to the exception cause as this method can be invoked with a ServiceNotFoundException or ResourceNotFoundException directly?

**Comment by sameera-jayasoma** [comment]
These changes are invalid now.

**Pull request closed without merging**